src/io/flutter/bazel/PluginConfig.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright  2017 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.bazel;
+
+import com.google.common.base.Objects;
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.util.Computable;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+ * An in-memory snapshot of the flutter.json file from a Bazel workspace.
+ */
+public class PluginConfig {
+  private final @NotNull Fields fields;
+
+  private PluginConfig(@NotNull Fields fields) {
+    this.fields = fields;
+  }
+
+  public @Nullable String getFlutterDaemonScript() {
+    return fields.flutterDaemonScript;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof PluginConfig)) return false;
+    final PluginConfig other = (PluginConfig)obj;
+    return Objects.equal(fields, other.fields);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(fields);
+  }
+
+  /**
+   * Reads plugin configuration from a file, if possible.
+   */
+  public static @Nullable
+  PluginConfig load(@NotNull VirtualFile file) {
+    final Computable<PluginConfig> readAction = () -> {
+      try {
+        final InputStreamReader input = new InputStreamReader(file.getInputStream(), "UTF-8");
+        final Fields fields = GSON.fromJson(input, Fields.class);
+        return new PluginConfig(fields);
+      } catch (IOException e) {
+        LOG.debug("failed to load flutter plugin config", e);
```

Comment:
It is LOG.error that writes stack trace to the log file and suggests user to submit it.
LOG.warn writes stack trace to log file silently.
LOG.info writes only message without stack to the log file.
LOG.debug by default does nothing at all unless you manually add corresponding category to Help | Debug Log Settings. With category added it writes message to the log file wihtout any stack trace.